src/app/services/transmart-resource/transmart-resource.service.ts

```diff
@@ -484,15 +484,38 @@ export class TransmartResourceService {
 
   // -------------------------------------- data table ---------------------------------------------
   getDataTable(tableState: TransmartTableState,
+               constraint: Constraint,
                offset: number, limit: number): Observable<TransmartDataTable> {
     const urlPart = `observations/table`;
-    const body = {
-      rows: tableState.rowDimensions,
-      columns: tableState.columnDimensions,
-      sort: tableState.sorting,
+    const highDims = ['assay', 'projection', 'biomarker', 'missing_value', 'sample_type'];
+    const rowDims = tableState.rowDimensions.filter((dim: string) => {
+      return !highDims.includes(dim);
+    });
+    const colDims = tableState.columnDimensions.filter((dim: string) => {
+      return !highDims.includes(dim);
+    });
+    let body = {
+      type: 'clinical',
+      constraint: constraint.toQueryObject(),
+      rowDimensions: rowDims,
+      columnDimensions: colDims,
       offset: offset,
       limit: limit
     };
+    if (tableState.rowDimensionSorting) {
+      let sort = [];
+      tableState.rowDimensionSorting.forEach((val, key) => {
+        sort.push([key, val]);
+      });
+      body['rowSort'] = sort;
```

Comment:
Here we need to add one exception - when there is only one rowSort dimension. The table endpoint accepts either a list of dimensions, example: `['study', 'patient']` (then by default the ascending order is applied) or list of tuples, example: `[['study','asc'], ['patient', 'desc']]`, so as implemented. Different situation is only for a single dimension, then it cannot be represented as  `['study', 'desc']`, because then 'desc' would be interpreted as a dimension name an an exception would accure.
For a single dimension case this parameter should be send as `['study': 'desc']` map pair.

Reply:
hmm, very bad format design for the api call, but I will change it